app/controllers/api/v1/hosts_controller.rb

```diff
@@ -125,11 +125,45 @@ def destroy
       eos
 
       def status
-        render :json => { :status => @host.host_status }.to_json if @host
+        Foreman::Deprecation.api_deprecation_warning('The /status route is deprecated, please use the new /status/configuration instead')
+        render :json => { :status => @host.get_status(HostStatus::ConfigurationStatus).to_label }.to_json if @host
+      end
+
+      api :GET, "/hosts/:id/status/:type", "Get specific status of host"
```

Comment:
I don't think there's any need to add new routes to v1, just make it available in v2 only.